drivers/sensor/sensirion/sts4x/sts4x.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2024 Jan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_STS4X_STS4X_H_
+#define ZEPHYR_DRIVERS_SENSOR_STS4X_STS4X_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/device.h>
+
+#define STS4X_CMD_RESET 0x94
+
+#define STS4X_RESET_TIME 1
+
+#define STS4X_CRC_POLY 0x31
+#define STS4X_CRC_INIT 0xFF
+
+struct sts4x_config {
+	struct i2c_dt_spec bus;
+	uint8_t repeatability;
+};
+
+struct sts4x_data {
+	uint16_t temp_sample;
+};
+
+static const uint8_t measure_cmds[3] = {0xE0, 0xF6, 0xFD};
+static const uint16_t measure_time_us[3] = {1600, 4500, 8300};
```

Comment:
Not enough here to justify having a separate private header file. Move the contents into the .c file